value.go

```diff
@@ -633,7 +640,13 @@ func assignValue(dst reflect.Value, src Value) error {
 			dst.SetUint(uint64(v))
 			return nil
 		default:
-			val = reflect.ValueOf(v)
+			switch dst.Type() {
+			case reflect.TypeOf(time.Time{}):
+				t := time.Unix(0, v).UTC()
```

Comment:
I don't know if we can make the assumption that `v` will be a number of nanoseconds here. Even if as a first step we only support encoding as nanosecond values, the decoding path shouldn't assume that all parquet files out there are meant to use nanoseconds only. We would have to scale the value based on the underlying unit of the timestamp type to properly interpret the integer.